lms/djangoapps/bookmarks/views.py

```diff
@@ -0,0 +1,301 @@
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:
+            usage_key = UsageKey.from_string(usage_id)
+        except InvalidKeyError as exception:
+            log.error(exception.message)
+            message = _(u"Invalid usage id")
+            return Response(
+                {
+                    "developer_message": message,
+                    "user_message": message
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        else:
+            # course_key may have an empty run property.
+            usage_key = usage_key.replace(course_key=modulestore().fill_in_run(usage_key.course_key))
+            course_key = usage_key.course_key
+
+        bookmarks_data = {
+            "usage_key": usage_key,
+            "course_key": course_key,
+            "user": request.user
+        }
+
+        try:
+            bookmark = Bookmark.create(bookmarks_data)
+        except ItemNotFoundError as exception:
+            log.error(exception.message)
+            return Response(
+                {
+                    "developer_message": u"Block with usage_id not found.",
+                    "user_message": _(u"Invalid usage id")
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        return Response(
+            BookmarkSerializer(bookmark, context={"fields": DEFAULT_FIELDS + OPTIONAL_FIELDS}).data,
+            status=status.HTTP_201_CREATED
+        )
+
+
+class BookmarksDetailView(APIView):
+    """
+    **Use Cases**
+
+        Get or Delete a specific bookmark.
+
+    **Example Requests**:
+
+        GET /api/bookmarks/v1/bookmarks/{username},{usage_id}/?fields=path&display_name
+
+        DELETE /api/bookmarks/v1/bookmarks/{username},{usage_id}/
+
+    **Response for GET**
+        Users can only get their own bookmarks
+
+        * id: String. The identifier string for the bookmark: {user_id},{usage_id}.
+
+        * course_id: String. The identifier string of the bookmark's course.
+
+        * usage_id: String. The identifier string of the bookmark's XBlock.
+
+        * display_name: (optional) String. Display name of the XBlock.
+
+        * path: (optional) List of dicts containing {"usage_id": "", display_name:""} for the XBlocks
+            from the top of the course tree till the parent of the bookmarked XBlock.
+
+        * created: ISO 8601 String. The timestamp of bookmark's creation.
+
+    **Response for DELETE**
+        Users can only delete their own bookmarks
+
+        A successful delete returns a 204 and no content.
+
+        Users can only delete their own bookmarks. If the requesting user
+        does not have username "username", this method will return with a
+        status of 404, Thee same is true in case if a bookmark does not exist.
+
+    """
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
```

Comment:
You should be able to use IsUserInUrl from /edx-platform/openedx/core/lib/api/permissions.py. To handle the 404 checking. Note that we changed to returning 403 if the user is staff instead of 404. This is true for mobile_api, profile_images, and preferences API.